src/RemoteInboxNotifications/ProductCountRuleProcessor.php

```diff
@@ -40,14 +40,10 @@ public function __construct( $product_query = null ) {
 	 * @return bool The result of the operation.
 	 */
 	public function process( $rule, $stored_state ) {
-		$products          = $this->product_query->get_products();
-		$new_product_count = property_exists( $stored_state, 'new_product_count' )
-			? $stored_state->new_product_count
-			: 0;
-		$count             = $products->total + $new_product_count;
+		$products = $this->product_query->get_products();
```

Comment:
Nice, thanks for making this fix! I thought it was a bit confusing. I was going to say to add a test for this change, but then I saw the existing tests and I think it's a simple enough logic to not require more tests.